From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Greg Price <greg@zulip.com>
Date: Tue, 24 May 2022 18:48:07 -0700
Subject: [upstream] rnav: Fix State tparam bound in NavigationProp

The state should be one that's valid for this param list,
not some other.
---
 types/@react-navigation/core/lib/typescript/src/types.js.flow | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git types/@react-navigation/core/lib/typescript/src/types.js.flow types/@react-navigation/core/lib/typescript/src/types.js.flow
index e3dd9a0d0..41f13ea18 100644
--- types/@react-navigation/core/lib/typescript/src/types.js.flow
+++ types/@react-navigation/core/lib/typescript/src/types.js.flow
@@ -202,7 +202,7 @@ export type NavigationContainerProps = {
 export type NavigationProp<
   ParamList: ParamListBase,
   RouteName: $Keys<ParamList> = $Keys<ParamList>,
-  State: NavigationState<> = NavigationState<ParamList>,
+  State: NavigationState<ParamList> = NavigationState<ParamList>,
   ScreenOptions: { ... } = { ... },
   EventMap: EventMapBase = EventMapBase,
 > = NavigationHelpersCommon<ParamList, State> & {
-- 
2.32.0

